src/shims/unix/socket.rs

```diff
@@ -300,15 +300,15 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
 
         // Expose peer file descriptor to each other.
         let weak_file_descriptor0 = file_descriptor0.downgrade();
-        let file_description0 = weak_file_descriptor0.upgrade().unwrap();
-        let mut fd_borrow0 = file_description0.borrow_mut();
-        let sv_file_description0 = fd_borrow0.downcast_mut::<SocketPair>().unwrap();
+        let mut file_description0 = file_descriptor0.borrow_mut();
+        let sv_file_description0 = file_description0.downcast_mut::<SocketPair>().unwrap();
         let weak_file_descriptor1 = file_descriptor1.downgrade();
-        let file_description1 = weak_file_descriptor1.upgrade().unwrap();
-        let mut fd_borrow1 = file_description1.borrow_mut();
-        let sv_file_description1 = fd_borrow1.downcast_mut::<SocketPair>().unwrap();
+        let mut file_description1 = file_descriptor1.borrow_mut();
+        let sv_file_description1 = file_description1.downcast_mut::<SocketPair>().unwrap();
         sv_file_description0.add_peer_fd(Rc::downgrade(&Rc::new(weak_file_descriptor1)));
         sv_file_description1.add_peer_fd(Rc::downgrade(&Rc::new(weak_file_descriptor0)));
+        drop(file_description0);
+        drop(file_description1);
```

Comment:
The manual drop is needed here because by using borrow_mut, ``file_description0`` seems to borrow ``file_descriptor0`` hence the compiler said it couldn't be moved when we need ``fds.insert_fd(file_descriptor0)``